### ISSUE TITLE
R Markdown Website not respecting language in _site.yml

### ISSUE THREAD

**Issue by daviddalpiaz** [rstudio/rmarkdown#1885]
The following `_site.yml` creates a website with an `<html>` tag that does not specify a language. (Accessibility checkers will throw a complaint about this.)

```
name: "My Website!"
output_dir: "docs"
output:
  html_document:
    lang: en
```

If within an individual .Rmd file a language is specified in its YAML header, the correct HTML is created.

For example, within the .Rmd you use:

```
---
title: "Title of Page"
author: "Cool R Markdown User"
lang: en
---
```

The rendered document then contains: `<html lang="en" xml:lang="en">`

Is this a feature, or a bug? (Perhaps it's good practice to specify a language for each page individually?)

- This question was originally asked here: https://community.rstudio.com/t/r-markdown-website-not-respecting-language-in-site-yml/77769

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by cderv** [comment]
Thanks for posting the link. I answered in the RStudio Community. 
https://community.rstudio.com/t/r-markdown-website-not-respecting-language-in-site-yml/77769/2

TL;DR: `lang` is not an option for the output format `html_document`. It should be set at the first level of the yaml. 
````yaml
name: "My Website!"
output_dir: "docs"
output: 
  html_document: default
lang: en
````
as you did within the Rmd file and Yaml header.

**Comment by cderv** [comment]
I'll reopen as currently the pandoc variable needs to be passed at a document level, and not a site level. Full reprex: https://github.com/daviddalpiaz/rmd-lang-test

**Comment by cderv** [comment]
Setting a pandoc variable for the all website outputs can be done using `pandoc_args` argument in `_site.yml` definition
```yaml
name: "my-website"
output:
  html_document:
    pandoc_args: !expr rmarkdown::pandoc_variable_arg("lang", "en")
navbar:
  title: "My Website"
  left:
    - text: "Home"
      href: index.html
    - text: "About"
      href: about.html
```

I think this is mainly due by the fact that `lang=en` should be in a yaml header processed by pandoc - and `_site.yml` is not. 

But more importantly I see this issue as a way to improve accessiblity 
> an <html> tag that does not specify a language. (Accessibility checkers will throw a complaint about this.)

It seems for accessibility compatibility, `lang` attribute in `<html>` should always be set. (ref: https://developer.paciellogroup.com/blog/2016/06/using-the-html-lang-attribute/) 

Maybe we could make one value the default for R markdown document ? Could be `en`, but obviously it would need to be change to another value if someone use rmarkdown. 

Anyway, this could obviously be more documented, and also maybe get better support to set it globally in rmarkdown website (by adding a `lang` argument in `html_document` as you tried ? 🤔 )

@jooyoungseo as you worked already a lot on accessibility, do you have some advice on this one ?

**Comment by jooyoungseo** [comment]
Thanks for copying me, @cderv.

> Maybe we could make one value the default for R markdown document ? Could be en, but obviously it would need to be change to another value if someone use rmarkdown.

I am a bit reluctant for us to set a default lang value. Forcing `lang` to a certain language (e.g., "en") by default can cause unexpected accessibility side-effects since screen readers' speech synthesizer is supposed to follow the `lang` attribute. For example, if an Rmd content is written in Korean and the creator forgot to change the `lang` value to "ko", a screen reader will switch its text-to-speech engine to the defined default `lang` (in this example, "en"), which is undesirable.

A safer way would be defaulting `<html lang = "">`. This does not force a certain language; however, this will be commonly recognized as client's system default locale.

I was working for Shiny `lang` attribute, and the related PR can be found in [rstudio/shiny#2920](https://github.com/rstudio/shiny/pull/2920).

**Comment by cderv** [comment]
Thanks @jooyoungseo ! 

To be clear, you are recommending for this attributes to be set in html produced by rmarkdown, aren't you ? 
Because currently, this attribute is not present in the html if `lang` is not set by the user in the yaml header.

**Comment by cderv** [comment]
This is related to #2062 regarding the lang attributes. 

This issue is also about how to set pandoc variables for all the document in a website.

**Comment by r-leyshon** [comment]
For anyone else that find these a priority, I've written a package that helps to avoid some of these 'gotchas' - [accessrmd](https://r-leyshon.github.io/access_rmd_docs/). 1.0.0 now on CRAN.
